src/main/java/software/aws/neptune/jdbc/utilities/QueryExecutor.java

```diff
@@ -217,9 +211,12 @@ private void resetQueryState() {
      *
      * @throws SQLException if query cancellation fails.
      */
-    public void cancelQuery() throws SQLException {
+    public void cancelQuery(final boolean isClosing) throws SQLException {
         synchronized (lock) {
             if (queryState.equals(QueryState.NOT_STARTED)) {
+                if (isClosing) {
+                    return;
+                }
```

Comment:
This part confuses me a bit but I could be lacking context.

If we're closing, then we only do nothing if the state is NOT_STARTED. What about CANCELLED, since we wouldn't care about an exception on closing? 

I'm not saying this is wrong, but it feels kind of convoluted and I want to double check that all checked paths are correct.

Reply:
We care about logging an exception on closing, but not if it wasn't starting.